dev/deploy-to-container/cli.js

```diff
@@ -0,0 +1,321 @@
+
+  // Pull latest DB image
+  console.info('Pulling DB docker image...')
+  const dbImagePullStream = await dock.pull('postgres:14.6')
+  await new Promise((resolve, reject) => {
+    dock.modem.followProgress(dbImagePullStream, (err, res) => err ? reject(err) : resolve(res))
+  })
+  console.info('Pulled DB docker image successfully.')
+  
+  // Pull latest Mail Archive Base image
+  console.info('Pulling latest Mail Archive base docker image...')
+  const appImagePullStream = await dock.pull('ghcr.io/ietf-tools/mailarchive-app-base:latest')
+  await new Promise((resolve, reject) => {
+    dock.modem.followProgress(appImagePullStream, (err, res) => err ? reject(err) : resolve(res))
+  })
+  console.info('Pulled latest Mail Archive base docker image.')
+
+  // Pull latest MQ image
+  console.info('Pulling latest MQ docker image...')
+  const mqImagePullStream = await dock.pull('rabbitmq:3')
+  await new Promise((resolve, reject) => {
+    dock.modem.followProgress(mqImagePullStream, (err, res) => err ? reject(err) : resolve(res))
+  })
+  console.info('Pulled latest MQ docker image.')
+
+  // Pull latest Celery image
+  // console.info('Pulling latest Celery docker image...')
+  // const celeryImagePullStream = await dock.pull('ghcr.io/ietf-tools/mailarchive-celery:latest')
+  // await new Promise((resolve, reject) => {
+  //   dock.modem.followProgress(celeryImagePullStream, (err, res) => err ? reject(err) : resolve(res))
+  // })
+  // console.info('Pulled latest Celery docker image.')
+
+  // Terminate existing containers
+  console.info('Ensuring existing containers with same name are terminated...')
+  const containers = await dock.listContainers({ all: true })
+  for (const container of containers) {
+    if (
+      container.Names.includes(`/ma-db-${branch}`) ||
+      container.Names.includes(`/ma-app-${branch}`) ||
+      container.Names.includes(`/ma-mq-${branch}`) ||
+      container.Names.includes(`/ma-celery-${branch}`) ||
+      container.Names.includes(`/ma-beat-${branch}`)
+      ) {
+      console.info(`Terminating old container ${container.Id}...`)
+      const oldContainer = dock.getContainer(container.Id)
+      if (container.State === 'running') {
+        await oldContainer.stop({ t: 5 })
+      }
+      await oldContainer.remove({
+        force: true,
+        v: true
+      })
+    }
+  }
+  console.info('Existing containers with same name have been terminated.')
+
+  // Get shared docker network
+  // console.info('Querying shared docker network...')
+  // const networks = await dock.listNetworks()
+  // if (!networks.some(n => n.Name === 'shared')) {
+  //   console.info('No shared docker network found, creating a new one...')
+  //   await dock.createNetwork({
+  //     Name: 'shared',
+  //     CheckDuplicate: true
+  //   })
+  //   console.info('Created shared docker network successfully.')
+  // } else {
+  //   console.info('Existing shared docker network found.')
+  // }
```

Comment:
Should not be commented. In the (very unlikely) case where mailarchive is the only app running on the dev sandbox, the deploy would fail.

Reply:
uncommented